catalog/task/add-bundles-to-image-index/0.1/add-bundles-to-image-index.yaml

```diff
@@ -0,0 +1,93 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: add-bundles-to-image-index
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+      Submit a build request to add operator bundles to an index image
+  params:
+    - name: IIB_URL 
+      description: IIB service url
+      type: string
+    - name: IIB_JSON_BUILD_ARGS
+      description: IIB JSON arguments to add a new build
+      type: string
+    - name: IIB_SECRET
+      type: string
+      description: Secret with iib credentials to be used
+  results:
+    - name: json_build_info
+      description: JSON build information for the requested build
+    - name: build_state
+      description: IIB Service build state
+  steps:
+    - name: add-bundles-to-image-index
+      image: quay.io/hacbs-release/release-utils
+      script: |
+        #!/usr/bin/env bash
+        #
+        dnf -y install krb5-workstation > /dev/null
+
+        # adds the json request parameters to a file to be used as input data
+        # for curl and preventing shell expansion.
+        json_input=/tmp/$$.tmp
+        cat > $json_input <<JSON
+        $(params.IIB_JSON_BUILD_ARGS)
+        JSON
+
+        # replaces {TOKEN} placeholder with credentials.
+        if [ -f /mnt/service-secret/registry_token ]; then
+            CREDS=$(cat /mnt/service-secret/registry_token)
+            sed -i "s/{TOKEN}/${CREDS}/g" ${json_input}
+        fi
+
+        # performs kerberos authentication.
+        export KRB5_CONFIG=/mnt/service-secret/krb5.conf
+        /usr/bin/kinit $(cat /mnt/service-secret/krb5_username) < /mnt/service-secret/krb5_password
+
+        # adds image to the index.
+        /usr/bin/curl -u : --negotiate -s -X POST -H "Content-Type: application/json" -d@${json_input} --insecure \
+        "$(params.IIB_URL)/builds/add" | jq | tee $(results.json_build_info.path)
+
+        # checks if the previous call returned an error.
+        ! jq -e -r ".error | select( . != null )" $(results.json_build_info.path)
+      volumeMounts:
+        - name: service-secret
+          mountPath: /mnt/service-secret
+    - name: wait-for-build-state
+      image: quay.io/hacbs-release/release-utils
+      script: |
+        #!/usr/bin/env bash
+        #
+        state=""
+        echo -en "waiting for build state to exit..."
+        while true; do
```

Comment:
we are relying on the overall tekton pipeline timeout here?

Reply:
I moved this back to in progress in Jira as I noticed that 2 requirements are missing. Including this.

Reply:
Ok, added the `IIB_BUILD_TIMEOUT_SECONDS` parameter.